### ISSUE TITLE
"File truncated" error on map with top layer blocks

### ISSUE THREAD

**Issue by 1AmYF** [yvt/openspades#812]
Joining a server with a map with lots of blocks on the upper layers (only hold by a few pillars) results in "File truncated" error. Joining with the classic 0.75 client works.
```
[C:\Users\tcpp\Documents\Projects\openspades\Sources\Gui\Runner.cpp:48] [!] Unhandled exception in SDLRunner:
File truncated
at C:\Users\tcpp\Documents\Projects\openspades\Sources\Client\GameMap.cpp:552
```
Example map to reproduce the error:
https://www.dropbox.com/s/ne1np6ntn6449tu/file_truncated_map.zip?dl=1

**Comment by feikname** [comment]
Related: #362

**Comment by NotAFile** [comment]
I think both of these should be fixed by the state packet fix?

**Comment by yvt** [comment]
This was caused by incorrect array bounds checks in the map loader, which were fixed in #830 when the map loader was partially rewritten.
